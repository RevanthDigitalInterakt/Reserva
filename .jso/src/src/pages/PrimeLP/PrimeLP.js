  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.default = undefined;
  var _asyncToGenerator2 = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[1]));
  var _slicedToArray2 = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[2]));
  var _react = _interopRequireWildcard(_$$_REQUIRE(_dependencyMap[3]));
  var _reactNative = _$$_REQUIRE(_dependencyMap[4]);
  var _testProps = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[5]));
  var _PrimeBenefits = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[6]));
  var _PrimeFAQ = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[7]));
  var _PrimeHero = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[8]));
  var _PrimeIntro = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[9]));
  var _PrimeSubscribe = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[10]));
  function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
  function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
  function PrimeLP(_ref) {
    var navigation = _ref.navigation;
    var _usePrimeInfo = (0, _$$_REQUIRE(_dependencyMap[11]).usePrimeInfo)(),
      onAddPrimeToCart = _usePrimeInfo.onAddPrimeToCart,
      isPrime = _usePrimeInfo.isPrime;
    var _useApolloFetchPolicy = (0, _$$_REQUIRE(_dependencyMap[12]).useApolloFetchPolicyStore)(['getFetchPolicyPerKey']),
      getFetchPolicyPerKey = _useApolloFetchPolicy.getFetchPolicyPerKey;
    var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2.default)(_useState, 2),
      loadingAddCartPrime = _useState2[0],
      setLoadingAddCartPrime = _useState2[1];
    var _usePrimeStore = (0, _$$_REQUIRE(_dependencyMap[13]).usePrimeStore)(['animationBag', 'handleClickContinue', 'isVisibleModalWelcome', 'changeStateAnimationBag', 'changeStateIsVisibleModalWelcome']),
      animationBag = _usePrimeStore.animationBag,
      handleClickContinue = _usePrimeStore.handleClickContinue,
      isVisibleModalWelcome = _usePrimeStore.isVisibleModalWelcome,
      changeStateAnimationBag = _usePrimeStore.changeStateAnimationBag,
      changeStateIsVisibleModalWelcome = _usePrimeStore.changeStateIsVisibleModalWelcome;
    var _useLandingPagePrimeQ = (0, _$$_REQUIRE(_dependencyMap[14]).useLandingPagePrimeQuery)({
        context: {
          clientName: 'gateway'
        },
        fetchPolicy: getFetchPolicyPerKey('landingPagePrime')
      }),
      rawData = _useLandingPagePrimeQ.data,
      loading = _useLandingPagePrimeQ.loading;
    var data = (0, _react.useMemo)(function () {
      return rawData == null ? undefined : rawData.landingPagePrime;
    }, [rawData == null ? undefined : rawData.landingPagePrime]);
    var handleOnModalHideSignIn = (0, _react.useCallback)(function () {
      changeStateAnimationBag(true);
    }, [changeStateAnimationBag]);
    var handleOnModalHide = (0, _react.useCallback)(function () {
      if (isPrime) {
        setTimeout(function () {
          changeStateIsVisibleModalWelcome(true);
        }, 500);
      }
    }, [changeStateIsVisibleModalWelcome, isPrime]);
    var onAddToCart = (0, _react.useCallback)(/*#__PURE__*/(0, _asyncToGenerator2.default)(function* () {
      try {
        if (!data || loadingAddCartPrime) return;
        if (isPrime) {
          navigation.popToTop();
          return;
        }
        setLoadingAddCartPrime(true);
        yield onAddPrimeToCart(false);
        handleOnModalHideSignIn();
      } catch (e) {
        _$$_REQUIRE(_dependencyMap[15]).ExceptionProvider.captureException(e);
      } finally {
        setLoadingAddCartPrime(false);
      }
    }), [data, handleOnModalHideSignIn, isPrime, loadingAddCartPrime, navigation, onAddPrimeToCart]);
    var onCloseModalWelcomePrime = (0, _react.useCallback)(function () {
      handleClickContinue();
      setTimeout(function () {
        navigation.navigate('Home');
      }, 300);
    }, [handleClickContinue, navigation]);
    return /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[16]).jsx)(_$$_REQUIRE(_dependencyMap[17]).SafeAreaView, {
      style: {
        backgroundColor: '#fff'
      },
      children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[16]).jsxs)(_$$_REQUIRE(_dependencyMap[18]).Box, {
        bg: "white",
        children: [/*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[16]).jsx)(_$$_REQUIRE(_dependencyMap[19]).TopBarDefaultBackButton, {
          loading: loading || loadingAddCartPrime,
          navigateGoBack: true
        }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[16]).jsx)(_$$_REQUIRE(_dependencyMap[20]).ModalBag, {
          isVisible: animationBag,
          onBackdropPress: function onBackdropPress() {
            return changeStateAnimationBag(false);
          },
          onModalHide: handleOnModalHide
        }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[16]).jsx)(_$$_REQUIRE(_dependencyMap[21]).ModalWelcomePrime, {
          isVisible: isVisibleModalWelcome,
          onClose: onCloseModalWelcomePrime
        }), !!(data && !loading) && /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[16]).jsxs)(_reactNative.ScrollView, Object.assign({
          contentContainerStyle: {
            paddingBottom: 100
          }
        }, (0, _testProps.default)('PrimeLP_page'), {
          children: [/*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[16]).jsx)(_PrimeHero.default, {
            data: data,
            onAddToCart: onAddToCart
          }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[16]).jsx)(_PrimeIntro.default, {
            data: data
          }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[16]).jsx)(_PrimeBenefits.default, {
            data: data
          }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[16]).jsx)(_PrimeSubscribe.default, {
            data: data,
            onAddToCart: onAddToCart
          }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[16]).jsx)(_PrimeFAQ.default, {
            data: data
          })]
        }))]
      })
    });
  }
  var _default = exports.default = PrimeLP;
