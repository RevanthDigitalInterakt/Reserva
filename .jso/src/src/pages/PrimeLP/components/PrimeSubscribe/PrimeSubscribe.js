  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.default = undefined;
  var _asyncToGenerator2 = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[1]));
  var _react = _interopRequireWildcard(_$$_REQUIRE(_dependencyMap[2]));
  var _reactNative = _$$_REQUIRE(_dependencyMap[3]);
  var _testProps = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[4]));
  var _EventProvider = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[5]));
  function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
  function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
  function PrimeSubscribe(_ref) {
    var data = _ref.data,
      onAddToCart = _ref.onAddToCart;
    var _usePrimeInfo = (0, _$$_REQUIRE(_dependencyMap[6]).usePrimeInfo)(),
      isPrime = _usePrimeInfo.isPrime;
    var onPressAddCart = (0, _react.useCallback)(/*#__PURE__*/(0, _asyncToGenerator2.default)(function* () {
      _EventProvider.default.logEvent('prime_press_add_to_cart_lp', {
        position: 'bottom'
      });
      onAddToCart();
    }), [onAddToCart]);
    return /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsxs)(_reactNative.View, Object.assign({}, (0, _testProps.default)('prime_subscribe_component'), {
      children: [/*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsxs)(_reactNative.View, {
        style: _$$_REQUIRE(_dependencyMap[8]).styles.wrapperTop,
        children: [/*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_$$_REQUIRE(_dependencyMap[9]).Typography, {
          variant: "descontoTag2",
          style: _$$_REQUIRE(_dependencyMap[8]).styles.title,
          children: "Assine Agora!"
        }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsxs)(_$$_REQUIRE(_dependencyMap[9]).Typography, {
          variant: "tituloSessao",
          style: _$$_REQUIRE(_dependencyMap[8]).styles.subtitle,
          children: [`Ganhe ${data.discountPercentage}% OFF* em todos os produtos da Reserva. Tudo isso por`, /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_$$_REQUIRE(_dependencyMap[9]).Typography, {
            variant: "precoPromocional2",
            style: _$$_REQUIRE(_dependencyMap[8]).styles.subtitleBold,
            children: ` ${data.installmentQty}x de R$ ${data.installmentPrice}/MÃªs.`
          })]
        })]
      }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_reactNative.ImageBackground, {
        source: _$$_REQUIRE(_dependencyMap[10]),
        resizeMode: "contain",
        style: _$$_REQUIRE(_dependencyMap[8]).styles.image,
        children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_$$_REQUIRE(_dependencyMap[9]).Typography, {
          style: _$$_REQUIRE(_dependencyMap[8]).styles.legalText,
          children: "*Exceto para parcerias, assinaturas, livros, drinks e m\xE1scaras. Desconto n\xE3o cumulativo com outras promo\xE7\xF5es."
        })
      }), /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_reactNative.TouchableOpacity, Object.assign({
        style: _$$_REQUIRE(_dependencyMap[8]).styles.button,
        onPress: onPressAddCart
      }, (0, _testProps.default)('prime_lp_bottom_button_add'), {
        children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[7]).jsx)(_$$_REQUIRE(_dependencyMap[9]).Typography, {
          style: _$$_REQUIRE(_dependencyMap[8]).styles.buttonText,
          children: isPrime ? 'CONTINUAR COMPRANDO' : 'Quero ser Prime'
        })
      }))]
    }));
  }
  var _default = exports.default = PrimeSubscribe;
