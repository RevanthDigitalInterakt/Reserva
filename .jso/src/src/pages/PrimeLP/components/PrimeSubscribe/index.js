  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function get() {
      return _PrimeSubscribe.default;
    }
  });
  var _PrimeSubscribe = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[1]));
