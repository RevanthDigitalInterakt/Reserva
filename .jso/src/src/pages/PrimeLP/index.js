  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  Object.defineProperty(exports, "default", {
    enumerable: true,
    get: function get() {
      return _PrimeLP.default;
    }
  });
  var _PrimeLP = _$$_REQUIRE(_dependencyMap[0])(_$$_REQUIRE(_dependencyMap[1]));
