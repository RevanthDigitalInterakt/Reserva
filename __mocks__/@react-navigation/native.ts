const navigate = jest.fn();

const useNavigation = () => ({
  navigate,
});

export {
  useNavigation,
};
