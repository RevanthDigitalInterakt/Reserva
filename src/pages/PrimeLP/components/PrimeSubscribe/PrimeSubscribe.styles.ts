import { StyleSheet } from 'react-native';
import configDeviceSizes from '../../../../utils/configDeviceSizes';
import { COLORS } from '../../../../base/styles/colors';

const commonWrapperStyle = StyleSheet.create({
  wrapper: {
    paddingHorizontal: 32,
    justifyContent: 'center',
    alignItems: 'center',
  },
});

export const styles = StyleSheet.create({
  wrapperTop: {
    ...commonWrapperStyle.wrapper,
    paddingTop: 32,
  },

  title: {
    fontSize: 20,
    lineHeight: 26,
    marginBottom: 24,
  },

  subtitle: {
    fontSize: 16,
    lineHeight: 20,
    color: COLORS.GRAY,
    marginBottom: 20,
    textAlign: 'center',
  },

  subtitleBold: {
    fontSize: 16,
    color: '#000000',
  },

  image: {
    ...commonWrapperStyle.wrapper,
    width: configDeviceSizes.DEVICE_WIDTH,
    paddingTop: 400,
    marginTop: -18,
  },

  legalText: {
    fontSize: 10,
    lineHeight: 14,
    textAlign: 'center',
    color: COLORS.GRAY,
    marginBottom: 24,
    top: 18,
  },

  button: {
    marginTop: 24,
    width: configDeviceSizes.DEVICE_WIDTH / 1.4,
    backgroundColor: '#000000',
    height: configDeviceSizes.DEVICE_HEIGHT / 13,
    alignItems: 'center',
    justifyContent: 'center',
    alignSelf: 'center',
  },

  buttonText: {
    textTransform: 'uppercase',
    color: '#ffffff',
    fontSize: 12,
    lineHeight: 14,
    letterSpacing: 0.5,
    fontWeight: '700',
  },
});
