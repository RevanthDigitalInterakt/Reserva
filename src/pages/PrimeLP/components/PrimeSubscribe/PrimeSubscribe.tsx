import React, { useCallback } from 'react';
import {
  View, TouchableOpacity, ImageBackground,
} from 'react-native';
import { Typography } from '@usereservaapp/reserva-ui';
import { styles } from './PrimeSubscribe.styles';
import testProps from '../../../../utils/testProps';
import type { PrimeDetailOutput } from '../../../../base/graphql/generated';
import EventProvider from '../../../../utils/EventProvider';
import { usePrimeInfo } from '../../../../hooks/usePrimeInfo';

const ImageProductSource = require('../../../../../assets/common/product.png');

interface IPrimeSubscribe {
  data: PrimeDetailOutput;
  onAddToCart: () => Promise<void>;
}

function PrimeSubscribe({ data, onAddToCart }: IPrimeSubscribe) {
  const { isPrime } = usePrimeInfo();

  const onPressAddCart = useCallback(async () => {
    EventProvider.logEvent('prime_press_add_to_cart_lp', { position: 'bottom' });
    onAddToCart();
  }, [onAddToCart]);

  return (
    <View {...testProps('prime_subscribe_component')}>
      <View style={styles.wrapperTop}>
        <Typography variant="descontoTag2" style={styles.title}>Assine Agora!</Typography>

        <Typography variant="tituloSessao" style={styles.subtitle}>
          {`Ganhe ${data.discountPercentage}% OFF* em todos os produtos da Reserva. Tudo isso por`}
          <Typography variant="precoPromocional2" style={styles.subtitleBold}>
            {` ${data.installmentQty}x de R$ ${data.installmentPrice}/Mês.`}
          </Typography>
        </Typography>
      </View>

      <ImageBackground source={ImageProductSource} resizeMode="contain" style={styles.image}>
        <Typography style={styles.legalText}>
          *Exceto para parcerias, assinaturas, livros, drinks e máscaras.
          Desconto não cumulativo com outras promoções.
        </Typography>
      </ImageBackground>

      <TouchableOpacity
        style={styles.button}
        onPress={onPressAddCart}
        {...testProps('prime_lp_bottom_button_add')}
      >
        <Typography style={styles.buttonText}>
          {isPrime ? 'CONTINUAR COMPRANDO' : 'Quero ser Prime'}
        </Typography>
      </TouchableOpacity>
    </View>
  );
}

export default PrimeSubscribe;
