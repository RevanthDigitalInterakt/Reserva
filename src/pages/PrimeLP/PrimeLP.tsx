import type { StackScreenProps } from '@react-navigation/stack';
import React, {
  useCallback, useMemo, useState,
} from 'react';
import { ScrollView } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';

import { useLandingPagePrimeQuery } from '../../base/graphql/generated';
import { ModalBag } from '../../components/ModalBag/ModalBag';
import { useApolloFetchPolicyStore } from '../../zustand/useApolloFetchPolicyStore';
import testProps from '../../utils/testProps';
import { ModalWelcomePrime } from '../../components/ModalWelcomePrime';
import { usePrimeInfo } from '../../hooks/usePrimeInfo';
import { TopBarDefaultBackButton } from '../../modules/Menu/components/TopBarDefaultBackButton';
import type { RootStackParamList } from '../../routes/StackNavigator';
import { usePrimeStore } from '../../zustand/usePrimeStore/usePrimeStore';
import PrimeBenefits from './components/PrimeBenefits';
import PrimeFAQ from './components/PrimeFAQ/PrimeFAQ';
import PrimeHero from './components/PrimeHero';
import PrimeIntro from './components/PrimeIntro';
import PrimeSubscribe from './components/PrimeSubscribe';
import { Box } from '../../components/Box/Box';
import { ExceptionProvider } from '../../base/providers/ExceptionProvider';

type IPrimeLP = StackScreenProps<RootStackParamList, 'PrimeLP'>;

function PrimeLP({ navigation }: IPrimeLP) {
  const { onAddPrimeToCart, isPrime } = usePrimeInfo();
  const { getFetchPolicyPerKey } = useApolloFetchPolicyStore(['getFetchPolicyPerKey']);
  const [loadingAddCartPrime, setLoadingAddCartPrime] = useState(false);

  const {
    animationBag,
    handleClickContinue,
    isVisibleModalWelcome,
    changeStateAnimationBag,
    changeStateIsVisibleModalWelcome,
  } = usePrimeStore([
    'animationBag',
    'handleClickContinue',
    'isVisibleModalWelcome',
    'changeStateAnimationBag',
    'changeStateIsVisibleModalWelcome',
  ]);

  const { data: rawData, loading } = useLandingPagePrimeQuery({
    context: { clientName: 'gateway' },
    fetchPolicy: getFetchPolicyPerKey('landingPagePrime'),
  });

  const data = useMemo(() => rawData?.landingPagePrime, [rawData?.landingPagePrime]);

  const handleOnModalHideSignIn = useCallback(() => {
    changeStateAnimationBag(true);
  }, [changeStateAnimationBag]);

  const handleOnModalHide = useCallback(() => {
    if (isPrime) {
      setTimeout(() => {
        changeStateIsVisibleModalWelcome(true);
      }, 500);
    }
  }, [changeStateIsVisibleModalWelcome, isPrime]);

  const onAddToCart = useCallback(async () => {
    try {
      if (!data || loadingAddCartPrime) return;

      if (isPrime) {
        navigation.popToTop();
        return;
      }

      setLoadingAddCartPrime(true);

      await onAddPrimeToCart(false);

      handleOnModalHideSignIn();
    } catch (e) {
      ExceptionProvider.captureException(e);
    } finally {
      setLoadingAddCartPrime(false);
    }
  }, [data, handleOnModalHideSignIn, isPrime, loadingAddCartPrime, navigation, onAddPrimeToCart]);

  const onCloseModalWelcomePrime = useCallback(() => {
    handleClickContinue();

    setTimeout(() => {
      navigation.navigate('Home');
    }, 300);
  }, [handleClickContinue, navigation]);

  return (
    <SafeAreaView style={{ backgroundColor: '#fff' }}>
      <Box bg="white">
        <TopBarDefaultBackButton
          loading={loading || loadingAddCartPrime}
          navigateGoBack
        />

        <ModalBag
          isVisible={animationBag}
          onBackdropPress={() => changeStateAnimationBag(false)}
          onModalHide={handleOnModalHide}
        />

        <ModalWelcomePrime
          isVisible={isVisibleModalWelcome}
          onClose={onCloseModalWelcomePrime}
        />

        {!!(data && !loading) && (
          <ScrollView
            contentContainerStyle={{ paddingBottom: 100 }}
            {...testProps('PrimeLP_page')}
          >
            <PrimeHero data={data} onAddToCart={onAddToCart} />

            <PrimeIntro data={data} />

            <PrimeBenefits data={data} />

            <PrimeSubscribe data={data} onAddToCart={onAddToCart} />

            <PrimeFAQ data={data} />
          </ScrollView>
        )}
      </Box>
    </SafeAreaView>
  );
}

export default PrimeLP;
