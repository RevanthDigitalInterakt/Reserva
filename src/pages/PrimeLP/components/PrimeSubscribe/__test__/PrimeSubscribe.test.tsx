import React from 'react';
import { ThemeProvider } from 'styled-components/native';
import { theme } from '@usereservaapp/reserva-ui';
import {
  fireEvent,
  render,
  screen,
} from '@testing-library/react-native';
import PrimeSubscribe from '../PrimeSubscribe';
import { mockPrimeData, addToCartMock } from '../../../../../../__mocks__/PrimeLP.mock';

const TestingComponent = (
  <ThemeProvider theme={theme}>
    <PrimeSubscribe data={mockPrimeData} onAddToCart={addToCartMock} />
  </ThemeProvider>
);

describe('PrimeSubscribe', () => {
  beforeEach(() => {
    render(TestingComponent);
  });
  it('should render properly and match with snapshot', () => {
    const primeSubscribe = screen.getByTestId('com.usereserva:id/prime_subscribe_component');

    expect(primeSubscribe).toBeOnTheScreen();
    expect(screen.toJSON()).toMatchSnapshot();
  });

  it('should call addPrimeToCart function', () => {
    const callToAction = screen.getByTestId(
      'com.usereserva:id/prime_lp_bottom_button_add',
    );

    fireEvent.press(callToAction);

    expect(addToCartMock).toHaveBeenCalledTimes(1);
  });
});
