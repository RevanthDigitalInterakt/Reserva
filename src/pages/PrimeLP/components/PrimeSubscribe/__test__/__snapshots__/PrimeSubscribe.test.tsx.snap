// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimeSubscribe should render properly and match with snapshot 1`] = `
<View
  accessibilityLabel="com.usereserva:id/prime_subscribe_component"
  testID="com.usereserva:id/prime_subscribe_component"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "paddingHorizontal": 32,
        "paddingTop": 32,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "ReservaSerif-Bold",
            "fontSize": 15,
          },
          Object {
            "fontSize": 20,
            "lineHeight": 26,
            "marginBottom": 24,
          },
        ]
      }
      variant="descontoTag2"
    >
      Assine Agora!
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Nunito-Regular",
            "fontSize": 15,
          },
          Object {
            "color": "#848484",
            "fontSize": 16,
            "lineHeight": 20,
            "marginBottom": 20,
            "textAlign": "center",
          },
        ]
      }
      variant="tituloSessao"
    >
      Ganhe 20% OFF* em todos os produtos da Reserva. Tudo isso por
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Nunito-Bold",
              "fontSize": 14,
            },
            Object {
              "color": "#000000",
              "fontSize": 16,
            },
          ]
        }
        variant="precoPromocional2"
      >
         12x de R$ 25/Mês.
      </Text>
    </Text>
  </View>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "marginTop": -18,
        "paddingHorizontal": 32,
        "paddingTop": 400,
        "width": 750,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "commons": Object {
            "GuideBermuda": [Circular],
            "GuideCalca": [Circular],
            "GuideCamiseta": [Circular],
            "GuideHoodie": [Circular],
            "GuideJaqueta": [Circular],
            "GuideMangaCurta": [Circular],
            "GuideMangaLonga": [Circular],
            "GuidePolo": [Circular],
            "GuideShort": [Circular],
            "GuideSueter": [Circular],
            "GuideSungaCueca": [Circular],
            "QrcodeBackground": [Circular],
            "americanExpress": [Circular],
            "arrowInstruction": [Circular],
            "bagEmpty": [Circular],
            "bannerCatalog": [Circular],
            "bannerLogin": [Circular],
            "bannerOffer": [Circular],
            "cardFront": [Circular],
            "cardVerso": [Circular],
            "clothingCare1": [Circular],
            "clothingCareBleach": [Circular],
            "clothingCareDrying": [Circular],
            "clothingCarePass": [Circular],
            "clothingCareProfessionalWashing": [Circular],
            "clothingCareWash": [Circular],
            "cupomModalBackground": [Circular],
            "dividerReserva": [Circular],
            "elo": [Circular],
            "emptyProductCatalog": [Circular],
            "facebook": [Circular],
            "foraDoAr": [Circular],
            "headerLogin": [Circular],
            "homeMock1": [Circular],
            "homeMock2": [Circular],
            "homeMock3": [Circular],
            "homeMock4": [Circular],
            "imageCashback": [Circular],
            "imageNotFound": [Circular],
            "instagram": [Circular],
            "localReserva": [Circular],
            "logoQRCode": [Circular],
            "mastercard": [Circular],
            "noStoresFound": [Circular],
            "noWishList": [Circular],
            "pinYou": [Circular],
            "purchaseConfirmation": [Circular],
            "raceImageBackground": [Circular],
            "saleOff": [Circular],
            "searchNotFound": [Circular],
            "selectRectangle": [Circular],
            "shirt1": [Circular],
            "shirt2": [Circular],
            "shirt3": [Circular],
            "shirt4": [Circular],
            "topBannerMock1": [Circular],
            "twitter": [Circular],
            "update": [Circular],
            "visa": [Circular],
            "whatsapp": [Circular],
            "withoutInternet": [Circular],
            "zoomHand": [Circular],
          },
          "default": Object {
            "GuideBermuda": [Circular],
            "GuideCalca": [Circular],
            "GuideCamiseta": [Circular],
            "GuideHoodie": [Circular],
            "GuideJaqueta": [Circular],
            "GuideMangaCurta": [Circular],
            "GuideMangaLonga": [Circular],
            "GuidePolo": [Circular],
            "GuideShort": [Circular],
            "GuideSueter": [Circular],
            "GuideSungaCueca": [Circular],
            "QrcodeBackground": [Circular],
            "americanExpress": [Circular],
            "arrowInstruction": [Circular],
            "bagEmpty": [Circular],
            "bannerCatalog": [Circular],
            "bannerLogin": [Circular],
            "bannerOffer": [Circular],
            "cardFront": [Circular],
            "cardVerso": [Circular],
            "chevronRight": [Function],
            "clothingCare1": [Circular],
            "clothingCareBleach": [Circular],
            "clothingCareDrying": [Circular],
            "clothingCarePass": [Circular],
            "clothingCareProfessionalWashing": [Circular],
            "clothingCareWash": [Circular],
            "cupomModalBackground": [Circular],
            "dividerReserva": [Circular],
            "elo": [Circular],
            "emptyProductCatalog": [Circular],
            "facebook": [Circular],
            "foraDoAr": [Circular],
            "headerLogin": [Circular],
            "homeMock1": [Circular],
            "homeMock2": [Circular],
            "homeMock3": [Circular],
            "homeMock4": [Circular],
            "imageCashback": [Circular],
            "imageNotFound": [Circular],
            "instagram": [Circular],
            "localReserva": [Circular],
            "logoPrime": [Function],
            "logoQRCode": [Circular],
            "mastercard": [Circular],
            "noStoresFound": [Circular],
            "noWishList": [Circular],
            "pinYou": [Circular],
            "purchaseConfirmation": [Circular],
            "raceImageBackground": [Circular],
            "saleOff": [Circular],
            "searchNotFound": [Circular],
            "selectRectangle": [Circular],
            "shirt1": [Circular],
            "shirt2": [Circular],
            "shirt3": [Circular],
            "shirt4": [Circular],
            "topBannerMock1": [Circular],
            "twitter": [Circular],
            "update": [Circular],
            "visa": [Circular],
            "whatsapp": [Circular],
            "withoutInternet": [Circular],
            "zoomHand": [Circular],
          },
          "svgs": Object {
            "chevronRight": [Function],
            "logoPrime": [Function],
          },
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": 750,
          },
          undefined,
        ]
      }
    />
    <Text
      style={
        Array [
          Object {},
          Object {
            "color": "#848484",
            "fontSize": 10,
            "lineHeight": 14,
            "marginBottom": 24,
            "textAlign": "center",
            "top": 18,
          },
        ]
      }
    >
      *Exceto para parcerias, assinaturas, livros, drinks e máscaras. Desconto não cumulativo com outras promoções.
    </Text>
  </View>
  <View
    accessibilityLabel="com.usereserva:id/prime_lp_bottom_button_add"
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#000000",
        "height": 102.61538461538461,
        "justifyContent": "center",
        "marginTop": 24,
        "opacity": 1,
        "width": 535.7142857142858,
      }
    }
    testID="com.usereserva:id/prime_lp_bottom_button_add"
  >
    <Text
      style={
        Array [
          Object {},
          Object {
            "color": "#ffffff",
            "fontSize": 12,
            "fontWeight": "700",
            "letterSpacing": 0.5,
            "lineHeight": 14,
            "textTransform": "uppercase",
          },
        ]
      }
    >
      Quero ser Prime
    </Text>
  </View>
</View>
`;
